### ISSUE TITLE
Handle pre-release versions in the doc actions

### ISSUE THREAD

**Issue by greschd** [pyansys/actions#217]
### 📝 Description of the feature

Currently, deploying a pre-release version (e.g. `0.3b1`) fails, see e.g. this pipeline run: https://github.com/pyansys/pydpf-composites/actions/runs/4565823726/jobs/8057628442

When deploying the documentation for a pre-release version, we should consider that the logic needs to be somewhat different:
- The `versions/stable` should _not_ be updated to point to the new pre-release
- The drop-down may contain the new version, but not marked as `(stable)`; maybe marked as `(pre-release)` instead
- If this minor version already exists, it should probably be ignored. This is more of a rare edge case, since typically there should not be beta releases at the patch level (e.g an existing `0.2.1`, and then a `0.2.2b1`).

**Comment by greschd** [comment]
Ping @jorgepiloto @Revathyvenugopal162 @RobPasMue for visibility.

**Comment by RobPasMue** [comment]
Yes, this would definitely be a nice-to-have. Thanks for posting the issue @greschd!

**Comment by jorgepiloto** [comment]
Thanks for suggesting this, @greschd. The actions fail because no wheelhouse artifacts got generated. Just add the [pyansys/actions/build-wheelhouse@v4](https://actions.docs.pyansys.com/version/stable/build-actions/index.html#build-wheelhouse-action) action.

Regarding multi-version for versions with suffix, this feature seems very interesting. I bet we can implement it in the actions.

**Comment by ludovicsteinbach** [comment]
Similarly it would be nice if it supported the "post" prefix

**Comment by jorgepiloto** [comment]
We had a short discussion about this issue. The logic for handling pre-releases would be as follow:

1. A pre-release (alpha, beta, rc, post, ...) is made.
2. The multi-version handles this version as usual.
3. If at some point, a stable release gets performed, any suffixed version for that version will no longer show, giving preference to the stable version.

**Comment by RobPasMue** [comment]
Yep let's go with that approach!

**Comment by jorgepiloto** [comment]
This issue is more complicated than what I expected. However, it is a desired feature.

I do not want to block new features that we want to implement in the incoming days. To protect the integrity of version 4.1, I am going to attach this issue to 4.2. This way, we can release 4.1 with simple features that are required, like #294.

**Comment by da1910** [comment]
As an unblocker for this, which remains a problem for us. Can we just get the docs-deploy-stable action to ignore pre-release versions of packages? That doesn't address the issue with post releases, but does mean that at least three sets of perfectly valid semantic versioning tags will work as "expected"?

We need to make use of alpha and beta tags as our openapi packages are built during API development for testing purposes, and this issue blocks either our publishing of builds, or our use of the documentation deployment action. The former is unsustainable, the latter is inconvenient.
